src/com/duckduckgo/mobile/android/activity/DuckDuckGo.java

```diff
@@ -863,7 +863,8 @@ public void feedItemSelected(FeedObject feedObject) {
     }
 
     public void feedItemSelected(String feedId) {
-        // TODO: fill in function body 
+        FeedObject feedObject = DDGApplication.getDB().selectFeedById(feedId);
+        feedItemSelected(feedObject);
```

Comment:
error: NULL_DEREFERENCE
object `feedObject` last assigned on line 866 could be null and is dereferenced by call to `feedItemSelected(...)` at line 867.

Reply:
Suggested fix: add null-pointer check to definition of `feedItemSelected(feedObject)` at [Line 843](https://github.com/septract/android-search-and-stories/blob/0f9ad25a8b19b1802f0a18d8a8f2a29fd0f56c1e/src/com/duckduckgo/mobile/android/activity/DuckDuckGo.java#L843): 
```
     public void feedItemSelected(FeedObject feedObject) {
+        if(feedObject==null) return;
         // keep a reference, so that we can reuse details while saving
         DDGControlVar.currentFeedObject = feedObject;	         
```

Reply:
[Inferbot] Apply suggested fix automatically by clicking here.

Reply:
[:shipit:](https://galois.com/) <- a clickable icon

Reply:
![Image of Yaktocat](https://octodex.github.com/images/yaktocat.png)